components/common/Tags.tsx

```diff
@@ -9,30 +10,52 @@ interface TagsProps {
 export default function Tags({ tags, margin = '', searchPath }: TagsProps) {
   return (
     <div className={`flex flex-wrap items-center gap-2.5 ${margin}`}>
-      {tags.map((tag) => (
-        <Tag key={tag} tag={tag} searchPath={searchPath} />
-      ))}
+      {searchPath
+        ? tags.map((tag) => (
+            <Link key={tag} href={`${searchPath}?tag=${tag}`} className="rounded-[1.875rem]">
+              <Tag tag={tag} hoverStyle="fill" />
+            </Link>
+          ))
+        : tags.map((tag) => <Tag key={tag} tag={tag} />)}
     </div>
   );
 }
 
+type HoverStyle = 'fill' | 'orange';
+type DefaultStyle = 'orange' | 'gray' | 'fill';
+
+const HOVER_STYLE: { [key in HoverStyle]: string } = {
+  fill: 'hover:bg-main-orange hover:border-main-orange hover:text-white',
+  orange: 'hover:bg-white hover:border-main-orange text-main-orange',
+};
+
+const DEFAULT_STYLE: { [key in DefaultStyle]: string } = {
+  orange: 'bg-white border-main-orange text-main-orange',
+  gray: 'bg-white border-neutral-400 text-neutral-400',
+  fill: 'bg-main-orange border-main-orange text-white',
+};
+
 interface TagProps {
   tag: string;
-  searchPath?: string;
+  hoverStyle?: HoverStyle;
+  defaultStyle?: DefaultStyle;
+  onClick?: (tag: string) => any;
```

Comment:
```suggestion
  onClick?: (tag: string) => void;
```

컴포넌트에서 반환값을 안 쓸 거라면 void, unknown, never 등을 추천드립니다!